### PR TITLE
Omega

### PR DESCRIPTION
**Pull request by MichaelPKennedy**
Major Changes:
-a mySQL migration script including new tables such as transactions and transactionDetails,  as well as some minor changes in other tables 

-login change-all users must sign in, admins sign in must include password
-register page- register with studentId

### CODE REVIEW

File: open_source_site/includes/erd-cart-datamodel-updated-4.png
Comment (reddign) [inline open_source_site/includes/erd-cart-datamodel-updated-4.png:1]: Is this file a duplicate of the one in the docs folder?  It makes more sense to me to be in the docs folder.  Can we delete this file?

File: open_source_site/includes/basket_users_transactions_ERD-4.mwb
Comment (reddign) [inline open_source_site/includes/basket_users_transactions_ERD-4.mwb:1]: Is this file a duplicate of the one in the docs folder?  It makes more sense to me to be in the docs folder.  Can we delete this file?

File: open_source_site/RegistrationProcess.php
Comment (reddign) [inline open_source_site/RegistrationProcess.php:1]: Is this file a copy of the file in web_src/classes?  If it is a file for the web interface, it makes more sense in the web_src area.  Can this file be deleted before merging into main?

**Review by reddign — COMMENTED** [review]
I'd like you to consider the extra files noted before approving.

Also if we allow this change into main the other teams will need to run the migration if they pull down from main.  I want to be sure everything is working before we allow these changes into main.  I will be looking at your commits in the Omega branch for this week's grade so there is no need to get your features into main at this time for grading.  However if they are complete I am all for it once a few duplicate files are addressed.

So I ran the migration script (migration-script-omega1.sql) to get the database changes. I did verify that the admin login still works but editing inventory link is now gone? Also Registration is just a button and doesn't go anywhere for me. Is the current state of your code?  Is there a need to merge this with main?  If so, just explain so we can understand and communicate with the class about the changes they will experience should they rebase from the main branch.  Thanks for all your hard work.  This really is quite impressive that your team is already at this point.

**Review by reddign — APPROVED** [review]
Thanks for cleaning up a few files.  Let's review the migration on Tuesday with the whole class.

File: data_src/api/user/create.php
Comment (ethanlaj) [inline data_src/api/user/create.php:25]: Can we keep the 400 status code if the student id is already taken?

File: data_src/api/user/create.php
Comment (ethanlaj) [inline data_src/api/user/create.php:34]: We should update the docs if we are changing the json returned upon success

File: docs/migration-script-omega1.sql
Comment (ethanlaj) [inline docs/migration-script-omega1.sql:1]: A migration implies that no data will be lost, but we are starting off by dropping the databaase. Can we instead replace Pantry.sql with this file?

**Review by ethanlaj — DISMISSED** [review]
See requested changes above

**Review by reddign — APPROVED** [review]
I agree that we need to keep the data_src/docs.html file up to date with the JSON changes that were made.  I'll be looking over this as the groups change over.  For now, I'll allow the merge and we will see what happens.

**Pull request merged**
